content/articles/2017-12-testthat-2-0.Rmd

```diff
@@ -0,0 +1,207 @@
+In addition, as part of a general process to make tidyverse and r-lib packages available more widely, we are now checking testthat with R 3.1, 3.2, 3.3, 3.4, and 3.5.
+
+## New default reporter
+
+A new default reporter, `ReporterProgress`, revamps the output to make use of colour and reveal details of failures as they occur:
+
+
+If you prefer the previous version, you can turn back time by setting `option(testthat.default_reporter = "summary")`.
+
+## Setup and teardown
+
+There are two new ways to setup and teardown code:
+
+*   New `setup()` and `teardown()` functions specify code to be run at the 
+    beginning and end of each test file. Write them next to each other so you
+    can easily confirm that each `setup()` is paired with a `teardown()` that
+    cleans up any changes to the global state.
+    
+    ```{r}
+    tmp <- tempfile()
+    setup({
+      writeLines("TEST DATA", tmp)
+    })
+    teardown({
+      unlink(tmp)
+    })
+    ```
+    
+*   `tests/testthat/setup-xyz.R` files are run before the first test file is 
+    executed. They are similar to the existing `helpers-xyz.R` files, but are 
+    not run by `devtools::load_all()`. Similarly, `test/teststhat/teardown-xyz.R` 
+    files are run after all tests are complete; use these to clean up any 
+    global changes made by the setup files.
+  
+## New and improved expectations
+
+We have identified a new family of expectations that compares the results of an expression to a known good value stored in a file. They are designed to be use in conjunction with git so that you can see what precisely has changed, and revert it if needed:
+
+*   `expect_known_output()` saves the output of an expression and will fail
+    the first time the output changes. The file is updated on each run, so
+    needs to be used in conjunction with git. It replaces `expect_output_file()`
+    which is now soft-deprecated.
+
+    ```{r, error = TRUE}
+    # File to save results: would usually be filename, which will be stored 
+    # in tests/testthat.
+    tmp <- tempfile()
+    
+    # The first run always succeeds, but warns
+    expect_known_output(mtcars[1:10, ], tmp, print = TRUE)
+    
+    # Subsequent runs will suceed only if the file is unchanged
+    # This will succeed:
+    expect_known_output(mtcars[1:10, ], tmp, print = TRUE)
+    
+    # This will fail
+    expect_known_output(mtcars[1:9, ], tmp, print = TRUE)
+    ```
+
+* `expect_known_value()` replaces `expect_equal_to_reference()`, which
+  has been soft-deprecated. It gains an update argument defaulting to `TRUE`.
+  This changes behaviour from the previous version, and soft-deprecated
+  `expect_equal_to_reference()` gets `update = FALSE`. 
+
+We've also improved tools for failures:
+
+* `expect_condition()` works like `expect_error()` but captures any
+  condition, not just error conditions.
+
+*  `expect_error()` gains a `class` argument that allows you to make an 
+  assertion about the class of the error object (#530).
```

Comment:
This reference to #530 seems to the sole issue tag of its kind. Is it intentional?